### DISCUSSION TITLE
Can paper size be customized like CUPS?

### DISCUSSION THREAD

**Discussion by iTanken** [sumatrapdfreader/sumatrapdf#3173] [General]
### CUPS Command-Line

- <https://openprinting.github.io/cups/doc/options.html>

> When Custom is listed for the PageSize option, you can specify custom media sizes using one of the following forms:
> 
> ```shell
> lp -o media=Custom.WIDTHxLENGTH filename
> lp -o media=Custom.WIDTHxLENGTHin filename
> lp -o media=Custom.WIDTHxLENGTHcm filename
> lp -o media=Custom.WIDTHxLENGTHmm filename
> ```
> 
> where "WIDTH" and "LENGTH" are the width and length of the media in points, inches, centimeters, or millimeters, respectively.

### SumatraPDF Command-Line

Can SumatraPDF customize the page size like CUPS:

```shell
SumatraPDF.exe -print-to-default .\test.pdf -print-settings noscale,1x,1,simplex,paper=Custom.100x200mm
```

Can paper size be customized like CUPS?

**Comment by GitHubRulesOK** [comment]
SumatraPDF uses windows defaults so paperkind can only be custom from the gui, 
however Windows default forms are settable and changeable so if preset and allowed for the printer you can use one or more custom form(s).

Much of this topic is discussed in the last printing enhancements
1st consider there is a related open issue that is a gotya for testing CLI behaviours so understand https://github.com/sumatrapdfreader/sumatrapdf/issues/1415
then review similar https://github.com/sumatrapdfreader/sumatrapdf/issues/972
then follow the reasoning in 
https://github.com/sumatrapdfreader/sumatrapdf/issues/1812

**Reply by iTanken** [reply]
Thank you. It would be nice if SumatraPDF could support adding custom forms automatically.

**Reply by GitHubRulesOK** [reply]
custom printer forms is a windows function and not always possible for all cases thus printing is as simple as GDI printing allows which is why when there are problems its usually down to windows update mods, user restrictions & limitations.

**Reply by iTanken** [reply]
OK, thanks again.
